### PR TITLE
Checking for missing generic type during annotation processing

### LINKED ISSUE #1131: Add a check in the annotation processor for the existence of the generic type of RealmLists
The annotation processor will crash if a model is defined like this:

```
private RealmList someField
```

### PR DESCRIPTION
**Pull request by kneth**
Closes #1131.

@emanuelez @cmelchior @nhachicha

### CODE REVIEW

File: realm-annotations-processor/src/main/java/io/realm/processor/ClassMetaData.java [hunk 1]
```diff
@@ -16,6 +16,7 @@
 
 package io.realm.processor;
 
+import java.lang.System;
```
Comment (cmelchior) [inline realm-annotations-processor/src/main/java/io/realm/processor/ClassMetaData.java]: Is this used by anything?

File: realm-annotations-processor/src/test/resources/some/test/MissingGenericType.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
+package some.test;
```
Comment (cmelchior) [inline realm-annotations-processor/src/test/resources/some/test/MissingGenericType.java]: Missing copyright

**Comment by cmelchior** [comment]
Apart from comments :+1:

**Comment by kneth** [comment]
Please review

@cmelchior @emanuelez @nhachicha

**Pull request merged**